src/sources/util/unix_stream.rs

```diff
@@ -71,7 +72,14 @@ pub fn build_unix_stream_source(
             let received_from: Option<Bytes> =
                 path.map(|p| p.to_string_lossy().into_owned().into());
 
-            let stream = socket.allow_read_until(shutdown.clone().map(|_| ()));
+            let stream = socket
+                .after_read(|byte_size| {
```

Comment:
I wonder if it makes sense to flip the ordering of this and the shutdown trigger. Like "read until shutdown" and _then_ "for each read, report the bytes read".

Reply:
it would not be possible because `AfterReadExt` would need `AllowReadUntil` to implement `AsyncRead` AND `AsyncWrite` but `AllowReadUntil` only implements `AsyncRead`.
Not sure if `AllowReadUntil` should implement `AsyncWrite`...